SDL/Event.cu

```diff
@@ -574,464 +564,154 @@ void SDL::Event::resetObjectsInModule()
-    cudaMemcpyAsync(hitsInGPU_event.zs,host_z,loopsize*sizeof(float),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU_event.rts,host_rts,loopsize*sizeof(float),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU_event.phis,host_phis,loopsize*sizeof(float),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU_event.etas,host_etas,loopsize*sizeof(float),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU_event.idxs,host_idxs,loopsize*sizeof(unsigned int),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU_event.moduleIndices,host_moduleIndex,loopsize*sizeof(uint16_t),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU_event.highEdgeXs,host_highEdgeXs,loopsize*sizeof(float),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU_event.highEdgeYs,host_highEdgeYs,loopsize*sizeof(float),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU_event.lowEdgeXs,host_lowEdgeXs,loopsize*sizeof(float),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU_event.lowEdgeYs,host_lowEdgeYs,loopsize*sizeof(float),cudaMemcpyHostToDevice,stream);
-    cudaMemcpyAsync(hitsInGPU_event.nHits,&loopsize,sizeof(unsigned int),cudaMemcpyHostToDevice,stream);// value can't correctly be set in hit allocation
-    cudaMemcpyAsync(hitsInGPU_event.hitRanges,module_hitRanges,nModules*2*sizeof(int),cudaMemcpyHostToDevice,stream);// value can't correctly be set in hit allocation
-    cudaMemcpyAsync(hitsInGPU_event.hitRangesLower,module_hitRangesLower,nModules*sizeof(int),cudaMemcpyHostToDevice,stream);// value can't correctly be set in hit allocation
-    cudaMemcpyAsync(hitsInGPU_event.hitRangesUpper,module_hitRangesUpper,nModules*sizeof(int),cudaMemcpyHostToDevice,stream);// value can't correctly be set in hit allocation
-    cudaMemcpyAsync(hitsInGPU_event.hitRangesnLower,module_hitRangesnLower,nModules*sizeof(int8_t),cudaMemcpyHostToDevice,stream);// value can't correctly be set in hit allocation
-    cudaMemcpyAsync(hitsInGPU_event.hitRangesnUpper,module_hitRangesnUpper,nModules*sizeof(int8_t),cudaMemcpyHostToDevice,stream);// value can't correctly be set in hit allocation
-cudaStreamSynchronize(stream);
+        if(modulesInGPU->subdets[lastModuleIndex] == Endcap && modulesInGPU->moduleType[lastModuleIndex] == TwoS)
+        {
+            int found_index = binary_search(geoMapDetId, iDetId, nEndCapMap);
+            float phi = 0;
+            // Unclear why these are not in map, but CPU map returns phi = 0 for all exceptions.
+            if (found_index != -1)
+                phi = geoMapPhi[found_index];
+            float cos_phi = cosf(phi);
+            hitsInGPU->highEdgeXs[ihit] = ihit_x + 2.5 * cos_phi;
+            hitsInGPU->lowEdgeXs[ihit] = ihit_x - 2.5 * cos_phi;
+            float sin_phi = sinf(phi);
+            hitsInGPU->highEdgeYs[ihit] = ihit_y + 2.5 * sin_phi;
+            hitsInGPU->lowEdgeYs[ihit] = ihit_y - 2.5 * sin_phi;
```

Comment:
All these float should be put in `2.5f` to avoid the conversion to doubles, otherwise switching to `cosf` and `sinf` probably doesn’t matter.